src/views/home/components/token-list/token-list.view.tsx

```diff
@@ -97,11 +62,12 @@ const TokenList: FC<TokenListProps> = ({ tokens, selected, chain, onSelectToken,
             }}
           />
           <div className={classes.list}>
-            {filteredTokens.slice(0, 20).map((token) => {
-              const isEnvToken = tokens.find((tkn) => tkn.address === token.address) !== undefined;
-              const isCustomToken =
-                getChainCustomTokens(chain).find((tkn) => tkn.address === token.address) !==
-                undefined;
+            {tokens.slice(0, 20).map((token) => {
+              const isImportedCustomToken = isChainCustomToken(token, chain);
```

Comment:
I don't agree that adding a property is a better option. Why add a property that can be guessed using a simple util?

Indeed, [we almost added this property in the past for a different reason](https://0xpolygon.slack.com/archives/C02FS3PHEGH/p1653049753593069), but luckily, back then, we find a way to avoid it.

Reply:
We agreed to leave this as-is for now.